### PR TITLE
Adds Mongo Express 

### PR DESCRIPTION
**Pull request by timheuer**
Addresses #1920 by adding `mongo-express` to the MongoDB extensions (with resource tests)

Some notes:
- Mongo Express cannot handle multiple hosts (https://github.com/mongo-express/mongo-express/issues/458) at this time so there is a diff express instance per mongodb instance (no shared admin)
- Basic auth is used on the admin portal using default creds for mongo-express (should this be configurable in the extension method?)
- Uses the endpoint port instead of the direct container port to handle reliable connection within the bridge docker network
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2091)

### CODE REVIEW

File: src/Aspire.Hosting/MongoDB/MongoExpressContainerResource.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.MongoDB;
+internal sealed class MongoExpressContainerResource : ContainerResource
+{
+    public MongoExpressContainerResource(string name) : base(name)
+    {
+    }
+}
```
Comment (davidfowl) [inline src/Aspire.Hosting/MongoDB/MongoExpressContainerResource.cs:12]:
```suggestion
internal sealed class MongoExpressContainerResource(string name) : ContainerResource(name)
{
}
```

File: src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs [hunk 2]
```diff
@@ -69,6 +70,45 @@ public static IResourceBuilder<MongoDBDatabaseResource> AddDatabase(this IResour
...
+                                  .ExcludeFromManifest();
+
+        return builder;
+    }
+
+    private static void ConfigureMongoExpressContainer(EnvironmentCallbackContext context, IResource resource)
+    {
+        var hostPort = GetResourcePort(resource);
+        
+        context.EnvironmentVariables.Add("ME_CONFIG_MONGODB_URL", $"mongodb://host.docker.internal:{hostPort}/?directConnection=true");
```
Comment (davidfowl) [inline src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs:98]: We're going to need to change all of this logic when we do remote resources (use mongo express on the cosmos mongo instance).

Comment (timheuer) [reply src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs:98]: Nothing to change now though right...we have this in a few resource types so something more universal/shared?

Comment (davidfowl) [reply src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs:98]: nope. we're good for now.

Comment (timheuer) [reply src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs:98]: Thanks...marked this ready for review

**Review by davidfowl — APPROVED** [review]

**Pull request merged**